### PR TITLE
#71806 Remove references to master branch

### LINKED ISSUE #71806: Migrate master to main: https://github.com/pytorch/pytorch

### PR DESCRIPTION
**Pull request by musebc**
This commit updates references to the master branch
so we can migrate it to the `main` branch.

Fixes #71806

### CODE REVIEW

File: .circleci/README.md [hunk 1]
```diff
@@ -104,7 +104,7 @@ All binaries are built in CircleCI workflows except Windows. There are checked-i
 
 Some quick vocab:
 
-* A \**workflow** is a CircleCI concept; it is a DAG of '**jobs**'. ctrl-f 'workflows' on https://github.com/pytorch/pytorch/blob/master/.circleci/config.yml to see the workflows.
```
Comment (janeyx99) [inline .circleci/README.md:107]: We actually might just want to delete this file. It's outdated and has not been updated in a while.

Comment (musebc) [reply .circleci/README.md:107]: Sounds good. I can do that.

Comment (musebc) [reply .circleci/README.md:107]: Deleted.

File: docs/source/fx.rst [hunk 2]
```diff
@@ -230,7 +230,7 @@ Graph Manipulation Examples
 -  `Conv/Batch Norm
    fusion <https://github.com/pytorch/pytorch/blob/40cbf342d3c000712da92cfafeaca651b3e0bd3e/torch/fx/experimental/optimization.py#L50>`__
 -  `replace_pattern: Basic usage <https://github.com/pytorch/examples/blob/master/fx/subgraph_rewriter_basic_use.py>`__
--  `Quantization <https://pytorch.org/docs/master/quantization.html#prototype-fx-graph-mode-quantization>`__
+-  `Quantization <https://pytorch.org/docs/main/quantization.html#prototype-fx-graph-mode-quantization>`__
```
Comment (janeyx99) [inline docs/source/fx.rst:233]: Has docs migrated to main yet?

File: docs/source/distributed.rst [hunk 3]
```diff
@@ -448,7 +448,7 @@ Note that you can use ``torch.profiler`` (recommended, only available after 1.8.
         tensor = torch.randn(20, 10)
         dist.all_reduce(tensor)
 
-Please refer to the `profiler documentation <https://pytorch.org/docs/master/profiler.html>`__ for a full overview of profiler features.
+Please refer to the `profiler documentation <https://pytorch.org/docs/main/profiler.html>`__ for a full overview of profiler features.
```
Comment (janeyx99) [inline docs/source/distributed.rst:451]: Has docs migrated yet?

**Review by janeyx99 — CHANGES_REQUESTED** [review]
I don't think you meant to commit the submodule (third_party) changes

**Comment by musebc** [comment]
Migrating the changes to #73037 to remove the `third_party` changes.

**Pull request closed without merging**